WeScan/Scan/ScannerViewController.swift

```diff
@@ -175,6 +178,82 @@ final class ScannerViewController: UIViewController {
+        guard let device = AVCaptureDevice.default(for: AVMediaType.video) else { return }
+        try device.lockForConfiguration()
+        
+        if device.isFocusPointOfInterestSupported && device.isFocusModeSupported(.autoFocus) {
+            device.focusPointOfInterest = point
+            device.focusMode = .autoFocus
+        }
+        
+        if device.isExposurePointOfInterestSupported, device.isExposureModeSupported(.continuousAutoExposure) {
+            device.exposurePointOfInterest = point
+            device.exposureMode = .continuousAutoExposure
+        }
+        
+        device.unlockForConfiguration()
+    }
+    
+    /// Resets the camera's exposure and focus point to automatic
+    private func resetFocusPoint() throws {
+        guard let device = AVCaptureDevice.default(for: AVMediaType.video) else { return }
+        try device.lockForConfiguration()
+        
+        if device.isFocusPointOfInterestSupported && device.isFocusModeSupported(.continuousAutoFocus) {
+            device.focusMode = .continuousAutoFocus
+        }
+        
+        if device.isExposurePointOfInterestSupported, device.isExposureModeSupported(.continuousAutoExposure) {
+            device.exposureMode = .continuousAutoExposure
+        }
+        
+        device.unlockForConfiguration()
+    }
+    
+    /// Removes an existing focus rectangle if one exists, optionally animating the exit
+    private func removeFocusRectangleIfNeeded(animate: Bool) {
+        guard let focusRectangle = focusRectangle else { return }
+        if animate {
+            UIView.animate(withDuration: 0.3, delay: 1.0, animations: {
+                focusRectangle.alpha = 0.0
+            }, completion: { (_) in
+                focusRectangle.removeFromSuperview()
+            })
+        } else {
+            focusRectangle.removeFromSuperview()
+        }
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        super.touchesBegan(touches, with: event)
+        
+        guard !CaptureSession.current.isAutoModeEnabled, let touch = touches.first else { return }
+        let touchPoint = touch.location(in: view)
+        let focusPoint: CGPoint = videoPreviewLayer.captureDevicePointConverted(fromLayerPoint: touchPoint)
+        
+        removeFocusRectangleIfNeeded(animate: false)
+        
+        focusRectangle = FocusRectangle(touchPoint: touchPoint)
+        view.addSubview(focusRectangle)
+        
+        defer {
+            removeFocusRectangleIfNeeded(animate: true)
+        }
+        
+        do {
+            try setTapToFocusPoint(focusPoint)
+        } catch {
+            let error = ImageScannerControllerError.inputDevice
+            guard let captureSessionManager = captureSessionManager else { return }
+            captureSessionManager.delegate?.captureSessionManager(captureSessionManager, didFailWithError: error)
+            return
+        }
+    }
```

Comment:
We're adding quite some logic to the `ScannerViewController`, making it responsible for more and more things. Would be great if we can migrate this part of the code to a `ScanFocusController` something like class, keeping this `ScannerViewController` smaller. 

That should also make it easier to test the code.

Reply:
Done! 399be14